src/main/mpack/common-services/SOLR/5.5.5/package/scripts/solr_utils.py

```diff
@@ -6,9 +6,11 @@
 from resource_management.core.shell import call
 from resource_management.libraries.functions.format import format
 
-COLLECTION_PATTERN = "\/solr\/[a-zA-Z0-9\._-]+"
+COLLECTION_PATTERN = "{solr_hdfs_directory}\/?[a-zA-Z0-9\._-]+"
 CORE_PATTERN = "{collection_path}\/core_node[0-9]+"
-WRITE_LOCK_PATTERN = "{0}/data/index/write.lock "
+INDEX_PATTERN = "{core_node_path}\/data\/index[\.0-9]{{0,20}}"
+WRITE_LOCK_PATTERN = "{0}/write.lock "
+HOSTNAME_VERIFIER_PATTERN = "{core_node_name}((?!shard|core_node).)*\"node_name\":\"{solr_hostname}"
```

Comment:
Just wanted to make a note from some of my testing that I don't think this regex quite works as-is.

A sample state.json looks like:

```
{"collection1":{
    "replicationFactor":"1",
    "shards":{
      "shard1":{
        "range":"80000000-ffffffff",
        "state":"active",
        "replicas":{"core_node1":{
            "core":"collection1_shard1_replica1",
            "base_url":"http://sandbox-hdp.hortonworks.com:8983/solr",
            "node_name":"sandbox-hdp.hortonworks.com:8983_solr",
            "state":"down",
            "leader":"true"}}},
      "shard2":{
        "range":"0-7fffffff",
        "state":"active",
        "replicas":{"core_node2":{
            "core":"collection1_shard2_replica1",
            "base_url":"http://sandbox-hdp.hortonworks.com:8983/solr",
            "node_name":"sandbox-hdp.hortonworks.com:8983_solr",
            "state":"down",
            "leader":"true"}}}},
    "router":{"name":"compositeId"},
    "maxShardsPerNode":"2",
    "autoAddReplicas":"false"}}
```

As I understand it, the `(?!shard|core_node).)` portion of the regex is intended to limit a potential match to the description of a single replica. That is, the intended match would look like:

```
"core_node2":{
            "core":"collection1_shard2_replica1",
            "base_url":"http://sandbox-hdp.hortonworks.com:8983/solr",
            "node_name":"sandbox-hdp.hortonworks.com:8983_solr",
            "state":"down",
            "leader":"true"}
```

But the word "shard' appears in the core name under normal circumstances (e.g. `collection1_shard2_replica1`) preventing any matches.